### ISSUE TITLE
Tensorflow 0.12 failed to install on Window 10

### ISSUE THREAD

**Issue by qiushi223** [tensorflow/tensorflow#6055]
I have installed python3.5.2+cuda8.0+cudnn5.1 on my computer with Windows 10(64-bits)+ GTX960M.Also excute the command line as following:
 
pip install --upgrade https://storage.googleapis.com/tensorflow/windows/gpu/tensorflow_gpu-0.12.0rc0-cp35-cp35m-win_amd64.whl

while when i get into python and input:    import tensorflow as tf , it shows errors:
![1](https://cloud.githubusercontent.com/assets/11583292/20858023/bc4712d2-b975-11e6-818f-b43b04c4f4ff.png)
![2](https://cloud.githubusercontent.com/assets/11583292/20858024/be9e4c8a-b975-11e6-8a3a-a590daab3afa.png)

Then i install the bazel but it is useless,what should i do?

**Comment by codeloverr** [comment]
Are you using 64-bit version of Python?

**Comment by marklit** [comment]
I can confirm I'm seeing the same error message on the 64-bit version of Python 3.5.2 on Windows 10.

```
Python 3.5.2 (v3.5.2:4def2a2901a5, Jun 25 2016, 22:18:55) [MSC v.1900 64 bit (AMD64)] on win32
```

I installed Tensor flow with the following command:

```
pip install --upgrade https://storage.googleapis.com/tensorflow/windows/gpu/tensorflow_gpu-0.12.0rc0-cp35-cp35m-win_amd64.whl
```

I'm using CUDA 8, cudnn-8.0, a GTX 1080 with the 369.30 driver.

**Comment by qiushi223** [comment]
@codeloverr I downloaded "python-3.5.2-amd64.exe" from https://www.python.org/downloads/release/python-352/
The description shows it is used "for AMD64/EM64T/x64, not Itanium processors",so i don't think the version of python is a problem.

**Comment by qiushi223** [comment]
@codeloverr YES!

**Comment by codeloverr** [comment]
I have successfully installed and used Tensorflow on Windows 10.
I am using GTX 1070 and 

- Python 3.5.2 64 -bit

- CUDA 8.0 & cuDNN for 8.0

After installing above mentioned enter following command in command prompt for CPU version
`pip install –upgrade https://storage.googleapis.com/tensorflow/windows/cpu/tensorflow-0.12.0rc0-cp35-cp35m-win_amd64.whl`

For GPU version
`pip install --upgrade https://storage.googleapis.com/tensorflow/windows/gpu/tensorflow_gpu-0.12.0rc0-cp35-cp35m-win_amd64.whl`

I'm having no issues with this.

**Comment by Carmezim** [comment]
@qiushi223 sorry it didn't work for you at first. Try seeing Issue #5949 which had a similar problem and see if the solution helps you.
Check if `MSVCP140.DLL` is in your %PATH% as in [here](https://github.com/tensorflow/tensorflow/pull/5955/commits/7db3b71559edec7e3a8b00f4511b473e3cea4052).

**Comment by marklit** [comment]
Installing Microsoft Visual C++ 2015 Redistributable Update 3 (x64 version) fixed it for me. Thanks.

**Comment by qiushi223** [comment]
@Carmezim @marklit It also works for me.Thanks a lot!

**Comment by pkxpp** [comment]
Installing Microsoft Visual C++ 2015 Redistributable Update 3 (x64 version) fixed it for me. Thanks.

**Comment by bhavsarpratik** [comment]
Hi, I have laid down step-by-step instructions to successfully install Tensorflow GPU on Windows.

https://github.com/bhavsarpratik/install_Tensorflow_GPU_windows

**Comment by Carmezim** [comment]
@bhavsarpratik That is nice you have a guide but I noticed you're are spamming issues with this and that is not cool. The official TensorFlow installation guide is complete and thorough also.

**Comment by bhavsarpratik** [comment]
@Carmezim Pardon me if this looks like spamming. I made the installation guide because I felt that some instructions were missing from the installation guide.

**Comment by Carmezim** [comment]
@bhavsarpratik you pasted the same message in at least four different issues, which is spamming. 
I think PRs are always welcome, feel free to submit one with what you think could be improved in the docs.

**Comment by drophit** [comment]
The problem was the cuDNN Library for me - for whatever reason cudnn-8.0-windows10-x64-v6.0 was NOT working - I used cudnn-8.0-windows10-x64-v5.1 - ALL GOOD!

My setup running Win10 with Nvidia GTX780M

- windows installed python 3.5.3-amd64 from [here](https://www.python.org/downloads/release/python-352/)
- cuDNN v5.1 for CUDA 8.0 from [here](https://developer.nvidia.com/rdp/cudnn-download)
- CUDA 8.0 x86_64 from [here](https://developer.nvidia.com/cuda-downloads)
- PATH vars set as expected to point at the cuDNN libs and python (added during python install)

**Comment by Carmezim** [comment]
@drophit TensorFlow only supports Python 64-bit on Windows.

**Comment by Bobe24** [comment]
@Carmezim 
Install Microsoft Visual C++ 2015 Redistributable Update 3 x64. works for me.
Thanks!

**Comment by atishsawant** [comment]
having the same issue as what was described above, but I've installed and uninstalled python, created the environment via conda and downloaded the visual studo 2015 .dll files and it still doesn't work. my cuda and cudnn files are also good. not sure what is going on? 



(tensorflow) C:\WINDOWS\system32>python
Python 3.5.4 |Continuum Analytics, Inc.| (default, Aug 14 2017, 13:41:13) [MSC v.1900 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow as tf
Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py", line 18, in swig_import_helper
    return importlib.import_module(mname)
  File "C:\ProgramData\Anaconda3\envs\tensorflow\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 985, in _gcd_import
  File "<frozen importlib._bootstrap>", line 968, in _find_and_load
  File "<frozen importlib._bootstrap>", line 957, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 666, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 577, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 938, in create_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
ImportError: DLL load failed: The specified module could not be found.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 41, in <module>
    from tensorflow.python.pywrap_tensorflow_internal import *
  File "C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py", line 21, in <module>
    _pywrap_tensorflow_internal = swig_import_helper()
  File "C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py", line 20, in swig_import_helper
    return importlib.import_module('_pywrap_tensorflow_internal')
  File "C:\ProgramData\Anaconda3\envs\tensorflow\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: No module named '_pywrap_tensorflow_internal'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\__init__.py", line 24, in <module>
    from tensorflow.python import *
  File "C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\__init__.py", line 49, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 52, in <module>
    raise ImportError(msg)
ImportError: Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py", line 18, in swig_import_helper
    return importlib.import_module(mname)
  File "C:\ProgramData\Anaconda3\envs\tensorflow\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 985, in _gcd_import
  File "<frozen importlib._bootstrap>", line 968, in _find_and_load
  File "<frozen importlib._bootstrap>", line 957, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 666, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 577, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 938, in create_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
ImportError: DLL load failed: The specified module could not be found.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 41, in <module>
    from tensorflow.python.pywrap_tensorflow_internal import *
  File "C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py", line 21, in <module>
    _pywrap_tensorflow_internal = swig_import_helper()
  File "C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py", line 20, in swig_import_helper
    return importlib.import_module('_pywrap_tensorflow_internal')
  File "C:\ProgramData\Anaconda3\envs\tensorflow\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: No module named '_pywrap_tensorflow_internal'


Failed to load the native TensorFlow runtime.

See https://www.tensorflow.org/install/install_sources#common_installation_problems

**Comment by atishsawant** [comment]
Here are my paths:

(tensorflow) C:\WINDOWS\system32>path
PATH=C:\ProgramData\Anaconda3\envs\tensorflow;C:\ProgramData\Anaconda3\envs\tensorflow\Library\mingw-w64\bin;C:\ProgramData\Anaconda3\envs\tensorflow\Library\usr\bin;C:\ProgramData\Anaconda3\envs\tensorflow\Library\bin;C:\ProgramData\Anaconda3\envs\tensorflow\Scripts;C:\ProgramData\Anaconda3;C:\ProgramData\Anaconda3\Library\mingw-w64\bin;C:\ProgramData\Anaconda3\Library\usr\bin;C:\ProgramData\Anaconda3\Library\bin;C:\ProgramData\Anaconda3\Scripts;C:\ProgramData\Oracle\Java\javapath;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v8.0\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v8.0\libnvvp;C:\Program Files (x86)\Intel\iCLS Client\;C:\Program Files\Intel\iCLS Client\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Skype\Phone\;C:\Program Files\Intel\WiFi\bin\;C:\Program Files\Common Files\Intel\WirelessCommon\;C:\Program Files (x86)\Windows Kits\8.1\Windows Performance Toolkit\;C:\Program Files\Microsoft SQL Server\110\Tools\Binn\;C:\Program Files (x86)\Microsoft SDKs\TypeScript\1.0\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;D:\Visual Studio\VC\bin;C:\Users\Atish Sawant\AppData\Local\Microsoft\WindowsApps, C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\bin;C:\Users\Atish Sawant\AppData\Local\Microsoft\WindowsApps

**Comment by atishsawant** [comment]
Fixed the issue, I had installed the cuDNN 7.0 from nVidia.
